contracts/AMBModule.sol

```diff
@@ -26,21 +26,32 @@ interface IAMB {
 }
 
 contract AMBModule {
+    event AmbModuleSetup(address indexed initiator, address indexed safe);
 
     IAMB public amb;
-    Executor public immutable executor;
+    Executor public executor;
     address public owner;
     bytes32 public chainId;
 
+    bool public isInitialized = false;
+
+    constructor() {
+        isInitialized = true;
```

Comment:
Same issue as the SafeDelay module. This will always cause setup to fail.
Your tests didn't pick this up because you were testing the mock contract which set it to false instead of true.